### PR TITLE
Feature/tnation/terraform 0.9

### LINKED ISSUE #172: Upgrade project to Terraform 0.9
See:

- https://www.terraform.io/upgrade-guides/0-9.html
- https://www.terraform.io/docs/backends/legacy-0-8.html
- https://www.terraform.io/docs/backends/

When configuring the S3 `backend`, I think that we should set `lock = false`.

### PR DESCRIPTION
**Pull request by tnation14**
## Overview
Updates Terraform to 0.9.x, which improves remote state management by adding a `backend` configuration object. Remote state is now managed via  `terraform init` instead `terraform remote config`.

Fixes #172

## Testing Instructions

 * Delete `deployment/terraform/.terraform`
 * Run `scripts/infra.sh plan`, ensure no changes are necessary.
```bash
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ export PFB_SETTINGS_BUCKET=staging-pfb-config-us-east-1
vagrant@vagrant-ubuntu-trusty-64:/vagrant$ ./scripts/infra.sh plan
No changes. Infrastructure is up-to-date.

This means that Terraform did not detect any differences between your
configuration and real physical resources that exist. As a result, Terraform
doesn't need to do anything.

```

### CODE REVIEW

**Comment by CloudNiner** [comment]
It looks like the build is broken, can this be prioritized this morning? We have a set of changes that need to go out so we can give a client access to the staging environment to begin testing.

Sorry I didn't catch this yesterday.

**Comment by tnation14** [comment]
No problem, working on it now.

**Comment by tnation14** [comment]
@CloudNiner just reverted the remote state; `develop` was just successfully deployed to staging. This PR is also blocked until  hashicorp/terraform#13149 is released.

File: scripts/infra [hunk 1]
```diff
@@ -71,8 +45,10 @@ if [ "${BASH_SOURCE[0]}" = "${0}" ]; then
 
                 BATCH_JOB_NAME_REVISION=$(docker-compose run --rm update-job-defs pfb-analysis-run-job.json "${PFB_AWS_ECR_ENDPOINT}/pfb-analysis:${GIT_COMMIT}")
                 popd
+                terraform init \
```
Comment (hectcastro) [inline scripts/infra]: In the other Terraform 0.9 upgrades, we explicitly removed the `.terraform` directory before executing `terraform init`.

**Comment by tnation14** [comment]
Updated.

**Comment by tnation14** [comment]
This is ready for another look.

**Pull request merged**